### PR TITLE
To stop seeding sample applications

### LINKED ISSUE #11714: [Server] Stop seeding sample Applications
### Current Behavior
Meshery Server loads sample applications at boot time. Applications are no longer a concept in Meshery. Applications have been replaced by Designs.

### Desired Behavior
Stop seeding sample Applications:

```
INFO[2024-08-20T15:48:45-05:00] extracting Applications from /Users/lee/.meshery/content/applications  app=meshery
INFO[2024-08-20T15:48:45-05:00] seeding sample Applications                   app=meshery
DEBU[2024-08-20T15:48:45-05:00] seeding Application: bookinfo                 app=meshery
DEBU[2024-08-20T15:48:45-05:00] seeding Application: emojivoto                app=meshery
DEBU[2024-08-20T15:48:45-05:00] seeding Application: httpbin                  app=meshery
DEBU[2024-08-20T15:48:45-05:00] seeding Application: imagehub                 app=meshery
```

---
### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

### PR DESCRIPTION
**Pull request by vaishnaviadhau**
**Notes for Reviewers**

This PR fixes #11714 
Screenshot 2024-09-04 145150

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
Thanks, @vaishnaviadhau. Although, the first item under Expected Behaviors has not been addressed.

**Comment by leecalcote** [comment]
> Thanks, @vaishnaviadhau. Although, the first item under Expected Behaviors has not been addressed.

@vaishnaviadhau, please either mark this PR as a partial fix or finish up the work here to make this a complete fix.

**Comment by vaishnaviadhau** [comment]
@leecalcote Apologies, I'll make the required changes

File: server/models/default_local_provider.go [hunk 1]
```diff
@@ -1546,7 +1508,9 @@ func getSeededComponents(comp string, log logger.Handler) ([]string, []string, e
 			return nil, nil, er
 		}
 	}
-	if !viper.GetBool("SKIP_DOWNLOAD_CONTENT") {
+	if comp == "Application" && viper.GetBool("SKIP_DOWNLOAD_APPLICATIONS") {
+		log.Info("Skipping seeding sample applications")
+	} else if !viper.GetBool("SKIP_DOWNLOAD_CONTENT") {
```
Comment (Yashsharma1911) [inline server/models/default_local_provider.go:1513]: @vaishnaviadhau we don't need to log that we are skipping seeding applications , you can remove it and also `SKIP_DOWNLOAD_APPLICATIONS` why we need this?

**Review by Yashsharma1911 — CHANGES_REQUESTED** [review]

**Comment by Yashsharma1911** [comment]
This issue needs to fix urgently, I'll push some changes on this PR

File: server/models/default_local_provider.go (hunk 1 above)
Comment (vaishnaviadhau) [reply server/models/default_local_provider.go:1513]: seeding of sample applications is being skipped, and the subsequent code to download and seed applications is not executed.

**Comment by vaishnaviadhau** [comment]
> This issue needs to fix urgently, I'll collaborate with you here by pushing some changes on this PR

I'm ready to collaborate and will push the necessary changes to the PR shortly.

File: server/cmd/main.go [hunk 2]
```diff
@@ -112,6 +112,7 @@ func main() {
 	viper.SetDefault(constants.ProviderENV, "")
 	viper.SetDefault("REGISTER_STATIC_K8S", true)
 	viper.SetDefault("SKIP_DOWNLOAD_CONTENT", false)
+	viper.SetDefault("SKIP_DOWNLOAD_APPLICATIONS", true)
```
Comment (Yashsharma1911) [inline server/cmd/main.go:115]:
```suggestion
```

**Review by Yashsharma1911 — APPROVED** [review]
Thankyou @vaishnaviadhau for taking up the issue and fixing it, go lint failure is happening because of other issues, this PR LGTM now.

**Comment by MUzairS15** [comment]
@Yashsharma1911 Dockerfile needs to be updated to stop bundling applications with the image.

**Pull request merged**